core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -43,23 +41,69 @@ impl StandardBroadcastRun {
         insert_shreds_elapsed: u64,
         broadcast_elapsed: u64,
         run_elapsed: u64,
+        clone_and_seed_elapsed: u64,
         num_entries: usize,
         num_shreds: usize,
         shred_index: u32,
+        slot: u64,
+        slot_ended: bool,
+        latest_shred_index: u32,
     ) {
+        self.stats.insert_shreds_elapsed += insert_shreds_elapsed;
+        self.stats.shredding_elapsed += shredding_elapsed;
+        self.stats.broadcast_elapsed += broadcast_elapsed;
+        self.stats.receive_elapsed += receive_entries_elapsed;
+        self.stats.clone_and_seed_elapsed += clone_and_seed_elapsed;
+
+        if slot_ended {
```

Comment:
I think we should also clear/report these metrics if a slot was interrupted before it could complete. For instance, in broadcast you could potentially be transmitting for slot 5, not finish the transmission, and start transmitting for slot 6. We don't want the metrics for slot 5 carrying over to slot 6. I added that check for an interrupted slot here: https://github.com/solana-labs/solana/pull/6269/files#diff-4399f6663fb60984424c7221c5046bd6R126

Reply:
Ok. Do you want to add that in your PR? Seems there is some extra logic there to add and that seems to be used for something else.

Reply:
yup thats fine!